### ISSUE TITLE
Everything is an NaN now

### ISSUE THREAD

**Issue by bigomega** [ruddfawcett/reposs#1]

**Comment by ruddfawcett** [comment]
Interesting...  I just tried it locally, and I get a non `NaN` result.  However, it doesn't seem to be working on Heroku -- and the GitHub API hasn't changed, and the function didn't break.

I'll look into this more and get back to you.

**Comment by bigomega** [comment]
Something's fishy in heroku. 
Yes, please get back when you get the chance 😊

**Comment by thatsIch** [comment]
Sadly still not working

**Comment by bogdanRada** [comment]
+1.  Would love to see this fixed. I use this in all my projects.  

I would gladly help solve this issue, but i think it's a problem with the server on heroku.com. I checked your code locally and works fine. Probably the server just needs to be restarted, maybe it crashed 

Or maybe is the fact that you don't send any header related to cache control as a response.  I think if you send this headers everything will start working again:

```
Cache-Control: no-cache, no-store, must-revalidate, max-age:0
 Expires: -1
```

you can see this https://github.com/github/markup/issues/224 for more information on that.

**Comment by bogdanRada** [comment]
i updated my last comment. Can you please check that ? I think will solve this problem. Thank you very much

**Comment by ruddfawcett** [comment]
I apologize for my negligence on this issue.  This should be fixed.  Thanks @bogdanRada!

**Comment by ruddfawcett** [comment]
This still seems to be an issue... 

It looks like the problem lies with Heroku.  I am using a `cedar-14` stack, and there are no errors in the logs.  Restarting the server with `heroku restart` works temporarily, but then `NAN` is returned again.  I'll look into it further.

**Comment by bogdanRada** [comment]
Yeah. I think i know why. Currently HTTP protocol on shields.io redirects to HTTPS. Can you try changing the url to HTTPS? I forgot about that

**Comment by bogdanRada** [comment]
The error happens because HTTP ons shields.io returns now Content-type text/html.

**Comment by ruddfawcett** [comment]
@bogdanRada switched to HTTPS -- still `NAN` :disappointed:.

**Comment by bogdanRada** [comment]
Thats strange :(

**Comment by bogdanRada** [comment]
I suspect why is happening. I had same issue on my repository http://github.com/bogdanRada/ruby-gem-downloads-badge. All my badges died when shields.io was switched on cloudflare which uses SNI for SSL connections.  Maybe  using curl instead of file_get_contents will fix the problem.

**Comment by bogdanRada** [comment]
Also i think you should see this discussion. It seems github introduces rate limiting. https://github.com/badges/shields/issues/529 and i think now your application will have to do autenticated requests to their API in order to work properly

**Comment by bogdanRada** [comment]
I think this will be solved by this pull request: https://github.com/ruddfawcett/reposs/pull/3 . Hope it helps :)

**Comment by ruddfawcett** [comment]
Closed by #3.
